networkx/readwrite/tests/test_gml.py

```diff
@@ -554,8 +554,8 @@ def assert_parse_error(gml):
             "directed 1 multigraph 1 ]"
         )
         assert_parse_error(
-            "graph [edge [ source u'u\4200' target u'u\4200' ] "
-            + "node [ id u'u\4200' label b ] ]"
+            "graph [edge [ source u'u\\4200' target u'u\\4200' ] "
+            + "node [ id u'u\\4200' label b ] ]"
```

Comment:
I'm not too sure what is even being tested here?

``` python
>>> gml = "graph [edge [ source wow target wow ] node [ id wow label lol ]]"
>>> nx.parse_gml(gml)._adj
{'lol': {'lol': {}}}
>>>  gml = "graph [edge [ source wow target wow ] node [ id wow label 䈀 ]]"
>>> nx.parse_gml(gml)._adj
...
...
NetworkXError: cannot tokenize 䈀 ]] at (1, 59)
```

Why can't a unicode character just be tokenised like a string? The GML parser doesn't seem right here?

This was added in https://github.com/networkx/networkx/pull/3603/

Reply:
> Why can't a unicode character just be tokenised like a string? The GML parser doesn't seem right here?

I don't know - maybe gml only supports a limited unicode character range (or ascii only)?

FWIW there's a slightly more informative chained exception with the following:

```python
# Taken directly from the updated test:
>>> gml = "graph [edge [ source u'u\\4200' target u'u\\4200' ] node [ id u'u\\4200' label b ] ]"
Traceback (most recent call last)
   ...
NetworkXError: cannot tokenize 'u\4200' target u'u\4200' ] node [ id u'u\4200' label b ] ] at (1, 23)

During handling of the above exception, another exception occurred:

Traceback (most recent call last)
   ...
NetworkXError: expected an int, float, string, '[' or string convertible ASCII value for node id or label, found 'u' at (1, 22)
```

Reply:
> I propose that we make this:   ...  '\u4200' as that is IIC a standard python way to represent the unicode character with code 4200. Does this make any sense? I am way out of my wheelhouse here. :}

Yes I think that's sounds good for now. I guess the first priority here is to fix the syntaxwarning.

Reply:
> I propose that we make this:   ...  '\u4200' as that is IIC a standard python way to represent the unicode character with code 4200. Does this make any sense? I am way out of my wheelhouse here. :}

The test is an `assert_parse_error` test, i.e. checking that the nx gml parser *fails* to read the line correctly... Funnily enough all three proposals (`u\4200`, `u\\4200` and `\u4200`) pass the test :upside_down_face: . So there likely is a deeper issue with the gml parser tokenization - but for the purposes of fixing the syntax warning, I switched to `\u4200` in e5e528c

Reply:
The GML technical doc explicitly says it does NOT support unicode (must be converted to html style 8859-1 chars like &amp).  So I think we want to put in valid unicode and check that it doesn't work.

Do we need the `u` in front of the single quotes?  I think we should probs remove them too (?? python2 historical cruft??).

Reply:
>  So I think we want to put in valid unicode and check that it doesn't work.

Sounds good - as of e5e528c the `\u4200` values are valid unicode characters!

> Do we need the u in front of the single quotes? I think we should probs remove them too (?? python2 historical cruft??).

Nope, those are no longer necessary either - removed in 62887bd